tests/draft2020-12/unevaluatedProperties.json

```diff
@@ -809,5 +809,223 @@
+                "data": {
+                    "bar": "test"
+                },
+                "valid": true
+            },
+            {
+                "description": "all in-place applicators are evaluated before unevaluatedProperties",
+                "data": {
+                    "bar": "test",
+                    "faz": "test"
+                },
+                "valid": true
+            }
+        ]
+    },
+    {
+        "description": "in-place applicator siblings, allOf has unevaluated",
+        "schema": {
+            "type": "object",
+            "allOf": [
+                {
+                    "properties": {
+                        "foo": true
+                    },
+                    "unevaluatedProperties": false
+                }
+            ],
+            "anyOf": [
+                {
+                    "properties": {
+                        "bar": true
+                    }
+                }
+            ]
+        },
+        "tests": [
+            {
+                "description": "base case: both properties present",
+                "data": {
+                    "foo": 1,
+                    "bar": 1
+                },
+                "valid": false
+            },
+            {
+                "description": "in place applicator siblings, bar is missing",
+                "data": {
+                    "foo": 1
+                },
+                "valid": true
+            },
+            {
+                "description": "in place applicator siblings, foo is missing",
+                "data": {
+                    "bar": 1
+                },
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "in-place applicator siblings, anyOf has unevaluated",
+        "schema": {
+            "type": "object",
+            "allOf": [
+                {
+                    "properties": {
+                        "foo": true
+                    }
+                }
+            ],
+            "anyOf": [
+                {
+                    "properties": {
+                        "bar": true
+                    },
+                    "unevaluatedProperties": false
+                }
+            ]
+        },
+        "tests": [
+            {
+                "description": "base case: both properties present",
+                "data": {
+                    "foo": 1,
+                    "bar": 1
+                },
+                "valid": false
+            },
+            {
+                "description": "in place applicator siblings, bar is missing",
+                "data": {
+                    "foo": 1
+                },
+                "valid": false
+            },
+            {
+                "description": "in place applicator siblings, foo is missing",
+                "data": {
+                    "bar": 1
+                },
+                "valid": true
+            }
+        ]
```

Comment:
Isn't this testing the same thing as "in-place application siblings, allOf has unevaluated". What's the benefit of test with the `unevaluatedProperties` in both places?

Reply:
See above -- if annotations from parent schemas are not properly set aside (which was the error in my implementation), one set of these tests could pass and the other could fail, depending on what keyword is evaluated first. (In my case, I evaluate allOf first, then anyOf, so annotations resulting from the allOf subschema will be seen by anyOf, but not the reverse.)